src/command_modules/azure-cli-registry/azure/cli/command_modules/registry/_format.py

```diff
@@ -0,0 +1,107 @@
+#---------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+#---------------------------------------------------------------------------------------------
+
+from collections import OrderedDict
+
+from ._utils import get_resource_group_by_registry
+
+def output_format(result):
+    '''Returns the list of container registries each of which is an ordered dictionary 
+    :param list/dict result: The container registry object(s) or deployment result(s)
+    '''
+    obj_list = result if isinstance(result, list) else [result]
+    return [_format(item) for item in obj_list]
+
+def _format(item):
+    '''Returns an ordered dictionary of the container registry or deployment result
+    :param dict item: The container registry object or deployment result
+    '''
+    if isinstance(item, dict) and 'id' in item and '/providers/Microsoft.Resources/deployments/' in item['id']:
+        return _format_deployment(item)
+    elif isinstance(item, dict) and 'id' in item and '/providers/Microsoft.Krater/registries/' in item['id']:
+        return _format_registry(item)
+    else:
+        raise ValueError('Unknown item: ' + str(item))
+
+def _format_deployment(item):
```

Comment:
It's okay, it makes sense to have it this way given the flexibility of the code. Thanks!